pypit/artrace.py

```diff
@@ -1073,12 +1077,15 @@ def new_find_objects(profile, bgreg, stddev):
     while np.any(gt_5_sigma & np.invert(_profile.mask)):
         # Find next maximum flux point
         imax = np.ma.argmax(_profile)
+        if not gt_5_sigma[imax]:
+            break
+        _profile[imax] = np.ma.masked  # Mask the peak
         # Find the valid source pixels around the peak
         f = np.arange(sz_x)[np.roll(not_gt_3_sigma, -imax)]
         # TODO: the ifs below feel like kludges to match old
         # find_objects function.  In particular, should objr be treated
         # as exclusive or inclusive?
-        objl[obj] = imax-sz_x+f[-1] if imax-sz_x+f[-1] > 0 else 1
+        objl[obj] = imax-sz_x+f[-1] if imax-sz_x+f[-1] > 0 else 0
```

Comment:
Yeah, I think the use of `... else 1` was me forcing the function to give exactly the same result as the cython fuction even though it makes more sense to be ` ... else 0`.  The same is true of the next line, where it makes more sense that it be `... else sz_x` instead of `... else sz_x-1`.  Would you mind testing if you still get the bad behavior without lines you added above?

Reply:
I fear that I can't, but I am willing to comment out those other
lines (leaving them in place) for now.  Shall we go with that?

ps.  I think  sz_x-1 is correct.

Reply:
On the three lines in question, I think this makes sense.  We should keep an eye on it though to make sure there's not some corner case that means we need those three lines.

On the `sz_x-1`, yeah, you're right.  I see the `objr[obj]+1` lines below it.  It would be worth checking how `objr` is used outside the function itself.  It's 1 pixel so we might not care, but if we ever slice an array with `[objl:objr]` instead of `[objl:objr+1]`, we're not catching that last pixel in the slice.

Approved!